lib/compress/zstd_compress.c

```diff
@@ -1203,42 +1204,78 @@ static size_t ZSTD_resetCCtx_usingCDict(ZSTD_CCtx* cctx,
                             U64 pledgedSrcSize,
                             ZSTD_buffered_policy_e zbuff)
 {
+    /* We have a choice between copying the dictionary context into the working
+     * context, or referencing the dictionary context from the working context
+     * in-place. We decide here which strategy to use. */
+    int attachDict = ( pledgedSrcSize <= 8 KB
```

Comment:
OK, so the in-place/copy threshold is applied here, irrespective of any compression level and strategy.

Given the relatively large differences in memory budget between strategies, I would expect the threshold to reflect that.

This could be classified as a later optimization.